src/build/arg/mod.rs

```diff
@@ -1903,8 +1921,16 @@ impl<'help> Arg<'help> {
     /// ```
     /// [options]: Arg::takes_value()
     /// [positional arguments]: Arg::index()
-    pub fn possible_values(mut self, names: &[&'help str]) -> Self {
-        self.possible_vals.extend(names);
+    pub fn possible_values(
+        mut self,
+        values: impl IntoIterator<Item = impl Into<ArgValue<'help>>>,
+    ) -> Self {
+        self.possible_vals.extend(
+            values
+                .into_iter()
+                .map(|value| value.into())
+                .collect::<Vec<ArgValue>>(),
```

Comment:
Why are we doing a `collect`?  We should be able to pass the iterator directly into `extend` without allocating a `Vec` first